heareval/predictions/task_predictions.py

```diff
@@ -1319,6 +1326,20 @@ def task_predictions(
             in_memory=in_memory,
         )
 
+        # Cache predictions for detailed analysis
+        if metadata["embedding_type"] == "event":
+            json.dump(
+                split_grid_points[i].predictor.test_predicted_events,
+                embedding_path.joinpath(f"{test_fold_str}.predictions.json").open("w"),
+                indent=4,
+            )
+        pickle.dump(
+            split_grid_points[i].predictor.test_predicted_labels,
+            open(
+                embedding_path.joinpath(f"{test_fold_str}.predicted-labels.pkl"), "wb"
+            ),
+        )
```

Comment:
a) We can't do any interesting downstream analysis unless predictions AND targets are pickled. This MUST be fixed.

b) Is it difficult to pickle these scores where the test stuff is actually computed, rather than relying upon nebulous global state that may change? (For b) maybe make an issue and move on.)

Reply:
By global state, I mean split_grid_points[i].predictor.test_predicted_labels. Maybe this is okay. but we still must fix a)

Reply:
Targets were already saved from before by the embedding module. @jorshi Thanks for adding it here as well. Will be in one place.

Reply:
For b) #340